### ISSUE TITLE
3.5a questions and comments 

### ISSUE THREAD

**Issue by nicm** [tmux/tmux#4162]
### Ask or discuss any quick questions about 3.5a here please.

**Do not use this issue for bug reports, they will be ignored** - please open a new issue with full information, see [CONTRIBUTING](https://github.com/tmux/tmux/blob/master/.github/CONTRIBUTING.md).

**[The 3.5a CHANGES file is here](https://raw.githubusercontent.com/tmux/tmux/3.5a/CHANGES).**

This is a bug fix release for 3.5, please also see this issue for the 3.5 release: https://github.com/tmux/tmux/issues/4136. The new changes in 3.5a are:

- Restore the `grey` colour name.

- Revert to using `/bin/sh` for `#()`, `run-shell` and `if-shell` - only popups change to use `default-shell`.

- Correct the sequences emitted for Shift keys and `C-BSpace` with extended keys enabled.

Please also note that 3.5 changes the way tmux represents keys, so some keys that were previously the same are now different. For example:

   - `C-A` and `C-a`, `C-B` and `C-b` and so on;
   - `C-i` and `Tab`;
   - `C-m` and `Enter`;
   - `C-@` and `C-Space`;
   - `C-Enter` and `Enter`;
   - `C-BTab` and `C-S-Tab`.
 
Any bindings for these keys will need to be changed in `.tmux.conf` for 3.5a.

Note that a fix in 3.5 and 3.5a to correctly set `SHELL` for `run-shell` and `if-shell` reveals a bug in the `tmux-sensible` script, causing it to set `default-shell` to `/bin/sh`. This can be worked around by adding this to `.tmux.conf` after the `tmux-sensible` script is invoked:
~~~~
set -gu default-command
set -g default-shell "$SHELL"
~~~~

**Comment by yasen-mclimate** [comment]
Hi - C-i still seems to be mapped to Tab, is there a workaround for this? Running tmux 3.5a in wezterm. The solution I was using to differentiate C-i from Tab no longer seems to be working after the recent update.

**Comment by blancpain** [comment]
Hi - C-i still seems to be mapped to Tab, is there a workaround for this? Running tmux 3.5a in wezterm. The solution I was using to differentiate C-i from Tab no longer seems to be working after the recent update.

**Comment by faelin** [comment]
This is a bit of a silly question, but, when I install tmux@3.5a through Homebrew (which definitely links to the 3.5a release), the `tmux -V` command gives `tmux 2.7`... Is the version tag supposed to be 2.7???

**Comment by nicm** [comment]
I expect you have 2.7 installed as well and are running that.

**Comment by faelin** [comment]
The command `which tmux` shows the command pointing correctly to Homebrew. The command `brew upgrade tmux` shows that Tmux is up to date: "Warning: tmux 3.5a already installed". But running `tmux -V` shows a version of `2.7`...

**Comment by rfrancoise** [comment]
I belatedly noticed (after pushing 3.5a to Debian backports) that something is not quite right with commit de6bce057a27fe3d1ac804c87cfe50846f75eb10, it breaks repeating box-drawing characters in `iftop`. The regression was already in 3.5.

With 3.5a:

<img width="521" alt="Screenshot 2024-10-09 at 10 32 14 PM" src="https://github.com/user-attachments/assets/ec622ece-3679-4d43-98d3-4ace3393fdfd">

With 3.4, or with 3.5a after reverting de6bce057a27fe3d1ac804c87cfe50846f75eb10:

<img width="515" alt="Screenshot 2024-10-09 at 10 31 24 PM" src="https://github.com/user-attachments/assets/4a258606-4331-41ea-985e-423661890b84">

Note that I actually start `iftop` with `TERM=xterm` to get it to display correctly under tmux in the first place. It wasn't an issue until now, and might not be related...

**Comment by nicm** [comment]
`TERM=tmux` and `screen` do not have `rep` so it probably will only be used with `TERM=xterm`, but come on... how can I do anything without logs? Open a proper issue please.

**Comment by rfrancoise** [comment]
> `TERM=tmux` and `screen` do not have `rep` so it probably will only be used with `TERM=xterm`, but come on... how can I do anything without logs? Open a proper issue please.

Of course, will do. The repro involves typing my user password, I need to do it differently and didn't have time to collect logs yesterday. Thanks!

**Comment by cmskog** [comment]
I have a bunch of mail related script under the combination "<prefix> C-m"(e.g. 'neomutt' under key combo "<prefix> C-m C-m").
However, that seems to complete stopped working under tmux 3.5a...

Noted that something regarding keys (C-m and Enter recognised as separate keys) were changed in 3.5a.
Might this be related ?

Regards

**Comment by x123** [comment]
> Note that a fix in 3.5 and 3.5a to correctly set `SHELL` for `run-shell` and `if-shell` reveals a bug in the `tmux-sensible` script, causing it to set `default-shell` to `/bin/sh`. This can be worked around by adding this to `.tmux.conf` after the `tmux-sensible` script is invoked:
> 
> ```
> set -g default-shell "$SHELL"
> ```

Hi, this doesn't seem to be the case when using nix-darwin, tmux-sensible, and tmux 3.5a in darwin. Even forcing the default shell as indicated still seems to default to `/bin/sh`. Could you clarify what bug exists in the tmux-sensible script or point me to the correct issue there if there is one? I don't see anything in the script that is obvious to me, nor do I see any open/closed issues on tmux-sensible that apply. Related nixpkgs issue (https://github.com/NixOS/nixpkgs/issues/348103) where we believe it's upstream (either tmux 3.5a or tmux-sensible, I guess).

**Comment by nicm** [comment]
Sorry, actually I think you need to override its `default-command` not `default-shell`.

**Comment by nicm** [comment]
I think this would probably be the best approach but someone will need to confirm it works:
~~~~
set -gu default-command
set -g default-shell "$SHELL"

**Comment by x123** [comment]
> I think this would probably be the best approach but someone will need to confirm it works:
> 
> ```
> set -gu default-command
> set -g default-shell "$SHELL"
> ```

Hi, I can confirm this works for me with tmux 3.5a, and tmux-sensible configured via home-manager. Just putting this into `extraConfig` and ensuring it's loaded after `tmux-sensible` seems to fix the issue. Thank you.

**Comment by zerochae** [comment]
Hello, I am leaving a comment here before opening an issue.
I am having an issue using NVM with tmux 3.5a.

The Node version is being fixed to system. 
However, if I pre-run `nvm use system`, I am able to change the Node version. Could you please check this issue?

- without `nvm use system` 
![Monosnap tmux 2024-10-16 16-03-12](https://github.com/user-attachments/assets/64410b23-c234-4b9e-8c38-4306bf4e4c22)

- with `nvm use system`

![Monosnap tmux 2024-10-16 16-03-18](https://github.com/user-attachments/assets/cf2fb4db-9722-4b6d-a3ca-176d91492a25)

**Comment by flipsi** [comment]
> Hi - C-i still seems to be mapped to Tab, is there a workaround for this? Running tmux 3.5a in wezterm. The solution I was using to differentiate C-i from Tab no longer seems to be working after the recent update.

I described a workaround [here](https://github.com/tmux/tmux/issues/4147#issuecomment-2416651207).

**Comment by nicm** [comment]
I don't think that is warranted really. Can't you just run your script with `/bin/sh` or use a shebang line?

**Comment by shuckster** [comment]
For anyone who had a config like this:

```
bind -r C-[ previous-window
bind -r C-] next-window
```

You need to change it to:

```
bind -r Escape previous-window
bind -r C-] next-window
```

**Comment by UsernameIsNotAvailableIs** [comment]
Hi,
after upgrading to 3.5a from an older 3.3 it seems there's a problem with urxvt (rxvt-unicode). Running tmux in it's default configuration and starting it, the prompt outputs:

`10;rgb:bebe/bebe/bebe]11;rgb:0000/0000/0000`

both local and also on remote sessions through SSH. This only happens in urxvt, not xterm as example. Changing the login shell for the user doesn't help, also purging .Xdefaults.

Running in logging mode and going through the server logfile:

```1729703880.174139 /dev/pts/3: wrote 480 bytes (of 480)
1729703880.174148 /dev/pts/3: waiting for redraw, 0 bytes left
1729703880.174196 /dev/pts/3: read 66 bytes (already 0)
1729703880.174210 /dev/pts/3: keys are 66 (\033[?1;2c\033[>85;95;0c\033]10;rgb:bebe/bebe/bebe\033\033]11;rgb:0000/0000/0000\033)
1729703880.174225 /dev/pts/3: received primary DA \033[?1;2c
1729703880.174234 applying terminal features: bpaste,focus,ignorefkeys,title
1729703880.174242 /dev/pts/3: \033[?7727h
1729703880.174251 /dev/pts/3: \033[m\017
1729703880.174258 /dev/pts/3: \033[?25h
1729703880.174265 /dev/pts/3: current mode ALL
1729703880.174271 /dev/pts/3: setting mode CURSOR
1729703880.174280 /dev/pts/3: \033[1;1H
1729703880.174289 /dev/pts/3: \033[1;53r
1729703880.174298 /dev/pts/3: complete key \033[?1;2c 0xfe000000000
1729703880.174309 /dev/pts/3: keys are 59 (\033[>85;95;0c\033]10;rgb:bebe/bebe/bebe\033\033]11;rgb:0000/0000/0000\033
.....
1729703880.189041 input_parse_buffer: %0 ground, 52 bytes: ^[]10;rgb:bebe/bebe/bebe^[^[]11;rgb:0000/0000/0000^[
1729703880.189054 screen_write_collect_end: 52 ^[]10;rgb:bebe/bebe/bebe^[^[]11;rgb:0000/0000/0000^[ (at 0,0)
1729703880.189084 /dev/pts/3: ^[]10;rgb:bebe/bebe/bebe^[^[]11;rgb:0000/0000/0000^[
```

Looks like the output I'm getting when running. If there's anything additional I may provide, feel free to ask and I'll take care asap.

Environment is FreeBSD, Xorg, urxvt, tmux 3.5a - already played with all the options the port provides (e.g. utf8 and so on).

**Comment by nicm** [comment]
This is a bug in rxvt-unicode, I'm not sure if it was fixed yet, maybe try a newer version.

**Comment by oriansj** [comment]
This line breaks with 3.5a but is 100% functional on 3.4:
bind-key -n C-T new-window

**Comment by yvf** [comment]
> For anyone who had a config like this:
> 
> ```
> bind -r C-[ previous-window
> bind -r C-] next-window
> ```
> 
> You need to change it to:
> 
> ```
> bind -r Escape previous-window
> bind -r C-] next-window
> ```

I also landed on this (but I'm using it as a prefix). Is this a bug, or a deliberate change ? It seems strange to now not accept the control character version.

**Comment by JamesonNetworks** [comment]
I'm going from Tmux 2.7 to Tmux 3.5a. Previously, I used to be able to do `Leader Key + Esc + Direction Arrow` to resize panes in tmux and as far as I know this didn't require any additional configuration (though I set it up many moons ago, so I may be misremembering). It appears on 3.5a this keyboard shortcut does not work and after spending a few hours trying to find the incantation to restore the behavior I am at a loss. 

I was hoping to appeal to those more versed in tmux in order to figure out how I can restore my workflow. Thanks in advance for any assistance for all the work you do and I appreciate the software!

**Comment by nicm** [comment]
You were probably managing to press Esc and the arrow keys within the default of escape time of 500 milliseconds, this has now been reduced. Try `set -s escape-time 500`.

**Comment by JamesonNetworks** [comment]
> You were probably managing to press Esc and the arrow keys within the default escape time of 500 milliseconds, this has now been reduced. Try `set -s escape-time 500`.

This was exactly it. Thank you! If there is anything I can do to help you out, let me know. I owe you one!

**Comment by lobopraveen** [comment]
I have the following in conf to avoid automatically copying and also exiting copy-mode on selection.

```
unbind -T copy-mode-vi MouseDragEnd1Pane
bind     -T copy-mode-vi MouseDragEnd1Pane send -X stop-selection
bind     -T copy-mode-vi MouseDown1Pane    select-pane
```

Is there a way to test if the selection is empty and exit the copy mode if empty? The trouble with above lines is that if I drag mouse in the empty space below the prompt, it enters copy-mode and selects empty space. It would be nice if I can test the selection and exit if empty. Even better if I can avoid entering copy mode if the selection is below the prompt.

**Comment by rbjorklin** [comment]
> Hi - C-i still seems to be mapped to Tab, is there a workaround for this? Running tmux 3.5a in wezterm. The solution I was using to differentiate C-i from Tab no longer seems to be working after the recent update.

Did you figure this out? I'm having the same problem using Alacritty.

Outside of Tmux:
```console
❯ showkey -a

Press any keys - Ctrl-D will terminate this program

^[[105;5u 	 27 0033 0x1b
 	 91 0133 0x5b
 	 49 0061 0x31
 	 48 0060 0x30
 	 53 0065 0x35
 	 59 0073 0x3b
 	 53 0065 0x35
 	117 0165 0x75
^D 	  4 0004 0x04
```

Whereas inside Tmux I get:
```console
❯ showkey -a

Press any keys - Ctrl-D will terminate this program

                  9 0011 0x09
^D        4 0004 0x04
```

No matter which permutation of `extended-keys <off,on,always>` and `extendend-keys-format <csi-u,xterm>` I use.

**Comment by ChuckFoo** [comment]
In earlier versions of tmux I used this binding to be able to drag reorder windows:
`bind-key -n MouseDrag1Status swap-window -d -t=`
but that no longer seems to work in 3.5a (I may have gone from 3.3 to 3.5a skipping 3.4, I should probably test that).

Anyone else out there use that binding as well can confirm/deny that it works/doesn't for them, as I realize it could just be a "me" problem.  I'm still digging in trying to diagnose, but thought I'd ask if anyone was seeing this too.

**Comment by joemonk** [comment]
I no longer seem to be able to edit paste buffers by pressing 'e'.
It seemed to happen overnight.

**Comment by nicm** [comment]
What does `tmux show -g editor` show and does command it shows work?

**Comment by joemonk** [comment]
> What does `tmux show -g editor` show and does command it shows work?

it showed /usr/bin/vi and no, it did not exist.

Fixed now, thanks. Is this something newly introduced? I haven't changed any config that I'm aware of.

**Comment by nicm** [comment]
`/usr/bin/vi` is the default, maybe you used to start tmux after setting `EDITOR` or `VISUAL` and now you don't?

**Comment by joemonk** [comment]
That makes a lot of sense. I have been starting tmux for a few years on system boot using resurrect+continuum plugins. I wonder if I had something else setting EDITOR before that service begins.
Glad to know that the issue was me though. 🧑‍🎓

**Comment by nuiuh** [comment]
> In earlier versions of tmux I used this binding to be able to drag reorder windows: `bind-key -n MouseDrag1Status swap-window -d -t=` but that no longer seems to work in 3.5a (I may have gone from 3.3 to 3.5a skipping 3.4, I should probably test that).
> 
> Anyone else out there use that binding as well can confirm/deny that it works/doesn't for them, as I realize it could just be a "me" problem. I'm still digging in trying to diagnose, but thought I'd ask if anyone was seeing this too.

I can confirm the issue.
It appeared in 3.4 and was already reported [in the 3.4 questions thread](https://github.com/tmux/tmux/issues/3836#issuecomment-1952824924).

**Comment by NovaViper** [comment]
Hey is `allow-passthrough` completely broken for anyone else? I'm trying to check if the option is being enabled but I notice when I run `tmux show-options -A`, it doesn't appear at all in the list, and no value appears when running `tmux show-options allow-passthrough`. I can confirm the issue is present on a bare config (nothing else but `set-option -g allow-passthrough on` in it). I'm on NixOS with tmux version 3.5a

**Comment by DHowett** [comment]
Terminal emulator maintainer here! Hi! I'm unsure of the etiquette on this repository, so I apologize if I've done something wrong and should have made a separate issue. This is a comment on 3.5a.

It's about `escape-time`. I've collapsed it so that it doesn't clutter up the repo. It's medium length and uses em-dashes, but I promise I didn't use a LLM to generate it.

<details>
<summary>Dustin rants about `escape-time`</summary>

We're getting a lot of user reports of broken escape sequences including those in bracketed paste--especially over network links--which point back to the new value of `escape-time` (10ms, introduced in 2e9d7ebf15615b51b014a12de03b0e5483aadf99 re #3844) being quite low.

I'm not sure what it is about our stack (windows, winsock, windows port of openssh, etc.) that triggers this more often, but... from a raw numbers perspective, I can't bring myself to agree with 10ms being a reasonable default.

I suspect that a lot of folks use tmux over a network.

Assuming a network with no additional latency, a 10ms gap is only sufficient to handle fragmented escape sequence reassembly at a distance of ~3000km. Considering that most networks have significantly more than 0ms additional latency, that range drops pretty quickly...

The fastest typist in the world (as of 2025? I think?) seems to be able to type 30 characters per second... with a delay of only 33ms between them. I understand that that person may want to press <kbd>Esc</kbd> as quickly as possible and then move on to their next character, but I am not sure that it's conceivable that that could happen within that 33ms budget.

Bringing the default up to 50ms increases the no-latency range to ~14,000km and I would love to meet the typist for whom that is an issue. 🙂 

Cross-linking: this was fixed for palette and DA1/2 reports in https://github.com/tmux/tmux/commit/e6bac234dbfbbd8709ac91f40dc9788e09daa028 and https://github.com/tmux/tmux/commit/cfb906a0ceb18846ac52561c5def88d155c5e54b.

Alternate solution: perhaps extending the escape time during bracketed paste would also work? It seems like adding a lot of workarounds for a bug that could be fixed more comprehensively... though it does allow users to set `escape-time` to however short they want and not risk breaking anything further!

</details>

**Comment by sigasigasiga** [comment]
Hello!

Thank you so much for tmux, I can't imagine my life without this tool!

When do you plan to release a new version? It's been a year since the latest release, and it seems to me that a plenty of features must have been committed already.

Personally, I've been eagerly waiting for CSI 2031 (#4269) to arrive in my distro. 

Thanks ❤️

**Comment by nicm** [comment]
Yeah will do a release soon, hope everyone has been testing master :-).

**Comment by LHurttila** [comment]
Any ideas why tmux doesn't load the config at ~/.config/tmux/tmux.conf automatically? Sourcing the config manually works.
Using Nixos 25.05 and  tmux is on version 3.5a. Only tmux plugins in use are power-theme, sensible and vim-tmux-navigator.

**Comment by nicm** [comment]
Is it finding a different config file first? What does `tmux display -p '#{config_files}'` say?

**Comment by LHurttila** [comment]
> Is it finding a different config file first? What does `tmux display -p '#{config_files}'` say?

/nix/store/45blr8ynrcj5l3dmclwgyl218yhl413z-hm_tmuxtmux.conf

Seems like some of the config is loaded but the parts of config set in programs.tmux.extraConfig of home manager's home.nix file are missing without manual sourcing on a new session.

**Comment by nicm** [comment]
Are you sure the packager isn't overriding `TMUX_CONF` when they build tmux so it doesn't look in `~/.config`?

**Comment by LHurttila** [comment]
> Are you sure the packager isn't overriding `TMUX_CONF` when they build tmux so it doesn't look in `~/.config`?

Not sure how I would check it . In any case  that would sound weird since some of the config is indeed loaded on session startup. 

All the configuration made in home.nix under programs.tmux ends up in  ~/.config/tmux/tmux.conf like I think it should. It could be that the problem is extraConfig being dumped on the end of the tmux.conf after plugins. Found a home manager issue that seems related (https://github.com/nix-community/home-manager/issues/3555)

**Comment by nicm** [comment]
Right but the default for tmux is to try `/etc/tmux.conf:~/.tmux.conf:$XDG_CONFIG_HOME/tmux/tmux.conf:~/.config/tmux/tmux.conf` and since it isn't finding `~/.config/tmux/tmux.conf` either that file doesn't actually exist or someone changed the configuration when they built tmux so it isn't checked.

**Comment by LHurttila** [comment]
Since
1. most of the configuration is loaded on session startup
2.  /etc/tmux.conf and ~/.tmux.conf do not exist
3.  ~.config/tmux/tmux.conf exists and contains all the configuration it is supposed to contain
4. disabling plugins removes them from the said config file and results in extraConfig getting sourced automatically on session startup

I'm inclined to believe this issue has to do with the ordering of the config sections by home manager options.
